express-api/src/typeorm/Migrations/1717022772567-NoteTypeTable.ts

```diff
@@ -0,0 +1,93 @@
+    await queryRunner.query(
+      `ALTER TABLE "project_note" RENAME COLUMN "note_type" TO "note_type_id"`,
+    );
+    await queryRunner.query(
+      `CREATE TABLE "note_type" ("created_by_id" uuid NOT NULL, "created_on" TIMESTAMP NOT NULL DEFAULT now(), "updated_by_id" uuid, "updated_on" TIMESTAMP DEFAULT now(), "id" SERIAL NOT NULL, "name" character varying(150) NOT NULL, "is_disabled" boolean NOT NULL DEFAULT false, "sort_order" integer NOT NULL DEFAULT '0', "description" text, "is_optional" boolean NOT NULL, "status_id" integer, CONSTRAINT "PK_c80a9fb97bd099d9e1da0b17d7a" PRIMARY KEY ("id"))`,
+    );
+    const user = await queryRunner.manager.findOne(User, {
+      where: { Username: 'system' },
+    });
+    const created_by_id = user.Id;
+    await queryRunner.query(
+      `INSERT INTO "note_type" (id, name, description, is_optional, created_by_id, status_id) VALUES 
+        (0, 'General', 'General default notes.', true, '${created_by_id}', NULL),
+        (1, 'Public', 'Publically shared notes.', true, '${created_by_id}', 4),
+        (2, 'Private', 'Private notes only visible to SRES.', true, '${created_by_id}', 4),
+        (3, 'Exemption', 'Exception notes.', true, '${created_by_id}', 15),
+        (4, 'AgencyInterest', 'ERP agency interest notes.', true, '${created_by_id}', 14),
+        (5, 'Financial', 'Financial notes.', true, '${created_by_id}', NULL),
+        (6, 'PreMarketing', 'Pre-marketing notes for SPL.', true, '${created_by_id}', 40),
+        (7, 'Marketing', 'Marketing notes for SPL.', true, '${created_by_id}', 41),
+        (8, 'ContractInPlace', 'Contract in place notes for SPL.', true, '${created_by_id}', 42),
+        (9, 'Reporting', 'Notes to include in reports.', true, '${created_by_id}', 4),
+        (10, 'LoanTerms', 'Loan term notes.', true, '${created_by_id}', NULL),
+        (11, 'Adjustment', 'Adjustment notes.', true, '${created_by_id}', 21),
+        (12, 'SplCost', 'SPL program cost notes.', true, '${created_by_id}', 21),
+        (13, 'SplGain', 'SPL gain or loss notes.', true, '${created_by_id}', 21),
+        (14, 'SalesHistory', 'Sales history notes.', true, '${created_by_id}', 21),
+        (15, 'CloseOut', 'Close out form notes.', true, '${created_by_id}', 21),
+        (16, 'Comments', 'General comments.', true, '${created_by_id}', 21),
+        (17, 'Appraisal', 'Appraisal note.', true, '${created_by_id}', 11),
+        (18, 'Offer', 'A purchaser has made an offer.', true, '${created_by_id}', 32),
+        (19, 'Remediation', 'A disposed project remediation note.', true, '${created_by_id}', 21),
+        (20, 'SplRemoval', 'Rational for being removed from SPL.', true, '${created_by_id}', 22),
+        (21, 'Documentation', 'Notes related to the provided documentation.', true, '${created_by_id}', NULL),
+        (22, 'ErpNotification', 'Notes displayed on ERP notifications.', true, '${created_by_id}', NULL) `,
+    );
```

Comment:
Putting all thoughts on the status id mapping here:
- **Exemption**: Are these notes for SRES to enter or clients? If clients, I think it would be id 8 "Submitted Exemption". If SRES, maybe 13? "Exemption Review"
- **ContractInPlace**:  Hard to know if this should be applicable to both 42 and 43 if we're using both.
- **Reporting**: I have no idea what this field is for. Is this just presented in reports and nowhere else?
- **CloseOut**: I think this one might come up at multiple points. It's definitely not restricted to just SPL.
- **Offer**: 32 (Disposed) could work as long as they only enter this info at the end of the workflow. I wonder if they ever enter the offer info before it's completed though.

I'm sure there will be some changes once notes start getting put on the page. Might have to fine-tune these as we go.

Reply:
Dunno if it answers your questions but for more context, I'm pretty sure that Exemption is supposed to be this field:
![image](https://github.com/bcgov/PIMS/assets/112989452/398df554-37bf-419c-a30d-8b450c856a5b)
Reporting is this field:
![image](https://github.com/bcgov/PIMS/assets/112989452/6994012a-5326-4a60-a426-0489dca646b6)

As for CloseOut, this field and pretty much all the other 21 status id fields are from that close out form. So you're right, the status isn't necessarily correct, it's kind of a special case. I wasn't sure where to put there.

Now Offer is a bit interesting. I thought it only appeared in the tab that reads Disposed, so I gave it Disposed status. But if you do a global search for "offersNote" in old PIMS you'll notice it actually appears in multiple places... So I guess it's valid under multiple statuses. Tying into your point about "ContractInPlace" too, maybe this means that we need another relational layer to these, a join table that can map note types to multiple statuses?